src/tests/read_only_mode.rs

```diff
@@ -1,41 +1,52 @@
 use crate::builders::CrateBuilder;
-use crate::{RequestHelper, TestApp};
+use crate::prelude::*;
 use diesel::prelude::*;
 
 #[test]
-fn can_hit_read_only_endpoints_in_read_only_mode() {
-    let (app, anon) = TestApp::init().empty();
-    app.db(set_read_only).unwrap();
-    anon.get::<()>("/api/v1/crates").assert_status(200);
+fn can_hit_read_only_endpoints_in_read_only_mode() -> CargoResult<()> {
+    let app = App::new();
+    app.set_read_only()?;
+    app.get("/api/v1/crates").send()?;
+    Ok(())
 }
 
 #[test]
-fn cannot_hit_endpoint_which_writes_db_in_read_only_mode() {
-    let (app, _, user, token) = TestApp::init().with_token();
+fn cannot_hit_endpoint_which_writes_db_in_read_only_mode() -> CargoResult<()> {
+    let app = App::new();
+    let user = app.create_user("new_user")?;
+    let token = app.token_for(&user)?;
+
     app.db(|conn| {
-        CrateBuilder::new("foo_yank_read_only", user.as_model().id)
+        CrateBuilder::new("foo_yank_read_only", user.id)
             .version("1.0.0")
-            .expect_build(conn);
-        set_read_only(conn).unwrap();
-    });
-    token
-        .delete::<()>("/api/v1/crates/foo_yank_read_only/1.0.0/yank")
-        .assert_status(503);
+            .build(conn)
+    })?;
+    app.set_read_only()?;
+    let resp = app
+        .delete("/api/v1/crates/foo_yank_read_only/1.0.0/yank")
+        .with_token(&token)
+        .send()
+        .allow_errors()?;
```

Comment:
Does this mean every time we test for an error, we have to call `allow_errors`?

Reply:
Right now, yeah. I'll likely end up with an `expect_status` function that returns the response if the status matched or panics otherwise. Erroring on 4xxx/5xx felt like the right default though, so this makes sense as a primitive to get out of that default behavior